packages/collector/test/tracing/frameworks/fastify/app.js

```diff
@@ -5,21 +5,92 @@
 
 'use strict';
 
-require('../../../../')();
+const semver = require('semver');
+const FASTIFY_VERSION = process.env.FASTIFY_VERSION || '1.14.6';
+const FASTIFY_REQUIRE = semver.major(FASTIFY_VERSION) === 1 ? 'fastify' : `fastify${semver.major(FASTIFY_VERSION)}`;
+
+const mock = require('mock-require');
 
+// NOTE: Link e.g. fastify2 to fastify or fastify3 to fastify
+if (FASTIFY_REQUIRE !== 'fastify') {
+  mock('fastify', FASTIFY_REQUIRE);
+}
+
+require('../../../../')();
 const fastify = require('fastify');
 
+// NOTE: preHandler got deprecated in v2 and removed in v3
+//       see https://github.com/fastify/fastify/pull/1750
+let handlerKey = 'beforeHandler';
+if ([2, 3].indexOf(semver.major(FASTIFY_VERSION)) !== -1) {
```

Comment:
> if (semver.major(FASTIFY_VERSION) !== 1)

Yeah good suggestion. Way more readable :) I'll change this line!

---

TBH I have not thought too much about it. I saw the warning and wanted to support both variants, because it was not reflected in our tests yet.

My thinking was:

v1 tests beforeHandler
v2 + v3 tests preHandler

With that we can ensure that it works for many cases? And on top of that I added more cases, which is the newest usage from the fastify docs.

I personally would keep the tests for now.

Reply:
Okay, then let's keep those tests. 👍